clang/test/SemaCXX/constexpr-return-non-void-cxx2b.cpp

```diff
@@ -0,0 +1,7 @@
+// RUN: %clang_cc1 -Wno-return-type -std=c++23 -fsyntax-only -verify %s
+// expected-no-diagnostics
+constexpr int f() { }
+static_assert(__is_same(decltype([] constexpr -> int { }( )), int));
```

Comment:
Should we emit at least a warning in case of missing return statement and non-void function and check it?

Reply:
For C++23, I think users will have to use `-Winvalid-constexpr` to be notified of "this will never be valid in a constant expression" thanks to P2448.

Reply:
@AaronBallman Should a new option (`-Winvalid-constexpr`) be added to control warnings for this case? Is that what you're referring to?

Reply:
@AaronBallman Thanks for the review. Should a new option (`-Winvalid-constexpr`) be added to control warnings for this case? Is that what you're referring to?

Reply:
We already have such an option, but it wasn't until today that the diagnostic worked in C++23: https://github.com/llvm/llvm-project/commit/e5f7123dfef3a80937d088d846ddb3b2bb1869b9

Should be nothing here for you to do.